src/main/java/com/microsoft/sqlserver/jdbc/SQLServerTrustManager.java

```diff
@@ -0,0 +1,166 @@
+ * connection property is set to true.
+ */
+final class PermissiveX509TrustManager implements X509TrustManager {
+    private final Logger logger;
+    private final String logContext;
+
+    PermissiveX509TrustManager(TDSChannel tdsChannel) {
+        this.logger = tdsChannel.getLogger();
+        this.logContext = tdsChannel.toString() + " (PermissiveX509TrustManager):";
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (logger.isLoggable(Level.FINER))
+            logger.finer(logContext + " Trusting client certificate (!)");
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (logger.isLoggable(Level.FINER))
+            logger.finer(logContext + " Trusting server certificate");
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return new X509Certificate[0];
+    }
+}
+
+
+/**
+ * This class implements an X509TrustManager that validates hostname.
+ *
+ * This validates the subject name in the certificate with the host name
+ */
+final class HostNameOverrideX509TrustManager implements X509TrustManager {
+    private final Logger logger;
+    private final String logContext;
+    private final X509TrustManager defaultTrustManager;
+    private String hostName;
+
+    HostNameOverrideX509TrustManager(TDSChannel tdsChannel, X509TrustManager tm, String hostName) {
+        this.logger = tdsChannel.getLogger();
+        this.logContext = tdsChannel.toString() + " (HostNameOverrideX509TrustManager):";
+        defaultTrustManager = tm;
+
+        // canonical name is in lower case so convert this to lowercase too.
+        this.hostName = hostName.toLowerCase(Locale.ENGLISH);
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (logger.isLoggable(Level.FINEST)) {
+            logger.finest(logContext + " Forwarding ClientTrusted.");
+        }
+
+        defaultTrustManager.checkClientTrusted(chain, authType);
+
+        // Explicitly validate the expiry dates
+        for (X509Certificate cert : chain) {
+            cert.checkValidity();
+        }
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (logger.isLoggable(Level.FINEST)) {
+            logger.finest(logContext + " Forwarding Trusting server certificate");
+        }
+
+        defaultTrustManager.checkServerTrusted(chain, authType);
+
+        // Explicitly validate the expiry dates
+        for (X509Certificate cert : chain) {
+            cert.checkValidity();
+        }
+
+        if (logger.isLoggable(Level.FINEST)) {
+            logger.finest(logContext + " default serverTrusted succeeded proceeding with server name validation");

```

Comment:
Nitpick: Capitalize D in default. Add comma after first clause (default serverTrusted succeeded).